src/encoded/types/target.py

```diff
@@ -80,13 +86,19 @@ def __name__(self):
 
     def _name(self, properties):
         root = find_root(self)
-        organism = root.get_by_uuid(self.organism(properties=properties, return_uuid=True))
-        organism_props = organism.upgrade_properties()
-        return u'{}-{}'.format(properties['label'], organism_props['name'])
+        organism_uuid = self.organism(properties=properties, return_uuid=True)
+        if organism_uuid is None:
+            source = properties['investigated_as'][0].replace(' ',  '_')
+        else:
+            organism = root.get_by_uuid(organism_uuid)
+            source = organism.upgrade_properties()['name']
+        return u'{}-{}'.format(properties['label'], source)
 
     def __resource_url__(self, request, info):
         request._linked_uuids.add(str(self.uuid))
         # Record organism uuid in linked_uuids so linking objects record
         # the rename dependency.
-        request._linked_uuids.add(str(self.organism(return_uuid=True)))
+        organism_uuid = self.organism(return_uuid=True)
+        if organism_uuid:
```

Comment:
I would probably use `if not organism_uuid` in the two checks above instead of `if organism_uuid is None` in order to make it consistent with this. I don't think you'll run into any problems by not checking specifically for None.

Reply:
Updated.